src/main/java/ch/uzh/ifi/hase/soprafs22/repository/GameRepository.java

```diff
@@ -12,14 +12,26 @@
 public interface GameRepository extends JpaRepository<Game, Long> {
     Game findById(long id);
 
-    @Query("select g from Game g where g.gameStatus = ch.uzh.ifi.hase.soprafs22.constant.GameStatus.ACTIVE " +
-            "and g.userId <> :userId " +
-            "and g not in (select g from Game g join Play p on g.id = p.gameId where p.userId = :userId)")
+
+    //Has to check both directions because user only wants to see cards from people that user is interested in and that are interested in user
+    @Query("select game from Game game where game.gameStatus = ch.uzh.ifi.hase.soprafs22.constant.GameStatus.ACTIVE " +
+            "and game.userId <> :userId " +
+            "and game not in (select g from Game g join Play p on g.id = p.gameId where p.userId = :userId)" +
+            "and game not in (select g from Game g join User u on g.userId=u.id where u.age < (select u.minAge from User u where u.id = :userId))" +
+            "and game not in (select g from Game g join User u on g.userId=u.id where u.age > (select u.maxAge from User u where u.id = :userId))" +
+            "and game not in (select g from Game g join User u on g.userId=u.id where u.minAge > (select u.age from User u where u.id = :userId))"+
+            "and game not in (select g from Game g join User u on g.userId=u.id where u.maxAge < (select u.age from User u where u.id = :userId))" +
+            "and game in(select g from Game g, User u1, User u2 where u1.id =:userId and u2.gender member of u1.genderPreferences and u1.gender member of u2.genderPreferences and u2.id = g.userId)")
     Page<Game> getOtherUserWithActiveGameThatWasNotPlayedOn(Pageable pageable, @Param("userId") long userId);
 
     @Query("select count (g) from Game g where g.gameStatus = ch.uzh.ifi.hase.soprafs22.constant.GameStatus.ACTIVE " +
             "and g.userId <> :userId " +
-            "and g not in (select g from Game g join Play p on g.id = p.gameId where p.userId = :userId)")
+            "and g not in (select g from Game g join Play p on g.id = p.gameId where p.userId = :userId)"+
+            "and g not in (select g from Game g join User u on g.userId=u.id where u.age < (select u.minAge from User u where u.id = :userId))" +
+            "and g not in (select g from Game g join User u on g.userId=u.id where u.age > (select u.maxAge from User u where u.id = :userId))" +
+            "and g not in (select g from Game g join User u on g.userId=u.id where u.minAge > (select u.age from User u where u.id = :userId))"+
+            "and g not in (select g from Game g join User u on g.userId=u.id where u.maxAge < (select u.age from User u where u.id = :userId))"+
+            "and g in(select g from Game g, User u1, User u2 where u1.id =:userId and u2.gender member of u1.genderPreferences and u1.gender member of u2.genderPreferences and u2.id = g.userId)")
```

Comment:
I would make this unidirectional: fetch the game from that user eventhough that user is not interested in you.

Reply:
I wouldn't as it logically doesn't make sense, as then we would have to check when rating white cards, if the white cards are played by users you are interested in.

Reply:
Why would you have to do that? Any dating app that I know shows you users that have no interest in you. And liking / disliking white cards is free anyway.

Reply:
No dating app shows you users if you are not in their preferences

Reply:
https://www.quora.com/Does-Tinder-only-show-profiles-on-people-whos-criteria-match-me Yes.

Reply:
But even then , it is just to attract more costumers and logically doesn't really make any sense to do this

Reply:
okey, for gender preferences it must be bidirectional, I see this. But for the rest unidirectional is enough, so it's good as it is now.